pkg/dbhash/dbhash.go

```diff
@@ -0,0 +1,120 @@
+package dbhash
+
+import (
+	"context"
+	"crypto/sha1"
+	"database/sql"
+	"encoding/hex"
+	"fmt"
+	"io"
+)
+
+// DatabaseStateHash calculates the hash of some state of the database according to the options passed.
+func DatabaseStateHash(ctx context.Context, tx *sql.Tx, opts ...Option) (string, error) {
```

Comment:
This is the method that the package exports to the external word making it possible to calculate the hash of some state of the database. I made it generic in the sense that you can pass the query for the data you want to calculate the state on. This decision was mostly so I could come up with some simple tests to make sure the logic of gathering the bytes and hashing was correct, not because I wanted to make a future-proof hashing solution (although i may help in that case as well)